src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -218,27 +218,41 @@ public void ValidateAtomicGuard(string id, NonPersistentDocumentFlags nonPersist
                     flags |= DocumentFlags.Resolved;
                 }
 
+                var revisionsCount = _documentDatabase.DocumentsStorage.RevisionsStorage.GetRevisionsCount(context, id);
+                if (revisionsCount > 0)
+                {
+                    flags |= DocumentFlags.HasRevisions;
```

Comment:
**The scenario I wanted to solve:**
Go into the studio.
Create a Doc.
Create a force-created revision.
Delete the doc.
create the doc again (a doc with the same id).
The new (revived) doc will have 2 revisions, but you won't be able to see them (because it wont have a 'HasRevisions' flag).

**this change is letting you to see them.**
look at the test `RevivedDocumentShouldHaveTheRevisionsOfTheDeletedDoc`.